cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/MetadataServiceModule.java

```diff
@@ -36,12 +36,21 @@
   protected void configure() {
     Multibinder<HttpHandler> handlerBinder = Multibinder.newSetBinder(
       binder(), HttpHandler.class, Names.named(Constants.Metadata.HANDLERS_NAME));
-
     CommonHandlers.add(handlerBinder);
     handlerBinder.addBinding().to(MetadataHttpHandler.class);
     handlerBinder.addBinding().to(LineageHTTPHandler.class);
     expose(Key.get(new TypeLiteral<Set<HttpHandler>>() { }, Names.named(Constants.Metadata.HANDLERS_NAME)));
+
     bind(MetadataAdmin.class).to(DefaultMetadataAdmin.class);
     expose(MetadataAdmin.class);
+
+    bind(ApplicationDetailFetcher.class).to(RemoteApplicationDetailFetcher.class);
+    expose(ApplicationDetailFetcher.class);
```

Comment:
I think you can bind and expose only `MetadataSubscriberService` in this module while keeping all the fetcher as private bindings.